pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -399,13 +405,33 @@ func (ds *Plugin) GetLatestAttestedNodeEventID(ctx context.Context) (eventID uin
 	return eventID, nil
 }
 
+func (ds *Plugin) CreateAttestedNodeEvent(ctx context.Context, event *datastore.AttestedNodeEvent) (*datastore.AttestedNodeEvent, error) {
+	return event, ds.withWriteTx(ctx, func(tx *gorm.DB) (err error) {
```

Comment:
As this creates db events, the passed struct will have an empty EventID, which GORM would then update based on the store in the database from the Autoincrement field that is used to supply the actual value.

This might explain why the same event if returned.  Going in, it is likely having an empty EventID, and coming out of "createAttestedNodeEvent(tx, event)" it likely has a filled EventID.

Reply:
Removed the return value, it just returns error now.